examples/semaphore-tmpl-level.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: semaphore-tmpl-level-
+spec:
+  entrypoint: semaphore-tmpl-level-example
+  templates:
+  - name: semaphore-tmpl-level-example
+    steps:
+    - - name: generate
+        template: gen-number-list
+    - - name: sleep
+        template: sleep-n-sec
+        arguments:
+          parameters:
+          - name: seconds
+            value: "{{item}}"
+        withParam: "{{steps.generate.outputs.result}}"
```

Comment:
Do we need a three template workflow to demonstrate template level semaphores? Can we simplify the example somehow?